rpc/auth.go

```diff
@@ -200,6 +200,60 @@ func MakeSimpleMultiAuthHandler(forEntities, expectedPayloads []string) AuthHand
 	})
 }
 
+// MakeSimpleMultiAuthPairHandler works similarly to MakeSimpleMultiAuthHandler with the addition of
+// supporting a key, id pair used to ensure that a key that maps to the id matches the key passed
+// during the function call.
+func MakeSimpleMultiAuthPairHandler(forEntities, expectedPayloads []map[string]string) AuthHandler {
+	if len(expectedPayloads) == 0 {
+		panic("expected at least one payload")
+	}
```

Comment:
[important] I believe this line is in the original `MakeSimpleMultiAuthHandler` above because before this PR, simple auth was the only form of auth, i.e. if there were nothing loaded into `MakeSimpleMultiAuthHandler`, there would be no way to control the robot, so it made sense to have the process panic. Now that we are having 2 different kinds of auth, I think this `panic` should only confirm if there is neither an entry in `MakeSimpleMultiAuthHandler` NOR in `MakeSimpleMultiAuthPairHandler`.

Reply:
aren't we only ever going to call one or the other though? my understanding was that each credential type only had one auth handler so we wouldnt be calling both. example from rdk: https://github.com/jr22/rdk/blob/55cab43ce0a2f5bf08caf929761afc5e750a6650/robot/web/web.go#L1042-L1044

or are you saying that the changes in rdk will be such that the api-key credential type will have a new type of auth handler that calls both?

Reply:
+1 to Julia's question. I assumed that we would be calling only one AuthHandler at a time either with a string or map of keyID and key?

Reply:
I think that as we are migrating over, it's possible to have both. We aren't going to delete secrets just because we added API keys. But we should eventually support removing the secrets.

Reply:
oh right because we need to support those legacy robot API keys and i guess there is no way to know when setting up the auth handler which one its gonna be

Reply:
I think I might have missed an in-person resolution to this commit. Did you all establish that it was OK to have the `panic` in both auth handlers?

Reply:
Yes, during the meeting we talked about how we would include branching logic in the rdk code path to use one of the handlers depending on the credential type in the setup hence the panic would confirm the absence of a payload in `MakeSimpleMultiAuthPairHandler` if we provide an empty map. In the case of branching logic, if our `handler.Config` [here](https://github.com/viamrobotics/rdk/blob/55cab43ce0a2f5bf08caf929761afc5e750a6650/robot/web/web.go#L1033-L1059) is called with the old api keys the we call `MakeSimpleMultiAuthHandler`, if it is called with the new api keys then we call, `MakeSimpleMultiAuthPairHandler` if it called with both then we call with some hybrid function that calls both in the background?

Reply:
kk thank you!